inference-engine/src/vpu/graph_transformer/src/stages/interpolate.cpp

```diff
@@ -0,0 +1,85 @@
+
+using namespace InferenceEngine;
+
+namespace vpu {
+
+void FrontEnd::parseInterpolate(const Model& model, const ie::CNNLayerPtr& _layer, const DataVector& inputs, const DataVector& outputs) const {
+    VPU_THROW_UNLESS(inputs.size() <= 4 && inputs.size() >= 1,
+                     "Interpolate stage with name {} must have no more than 4 inputs and no less than 1 input, actually provided {} inputs",
+                     _layer->name, inputs.size());
+
+    VPU_THROW_UNLESS(outputs.size() == 1,
+                     "Interpolate stage with name {} must have only 1 output, actually provided {} outputs",
+                     _layer->name, outputs.size());
+
+    using ngraph::opset4::Interpolate;
+    const auto interpolateMode = _layer->GetParamAsString("mode");
+
+    const auto input = inputs[0];
+    const auto output = outputs[0];
+
+    // try to use existing resize layers
+    if (input->desc().dimsOrder() == DimsOrder::NCHW || input->desc().dimsOrder() == DimsOrder::NHWC ||
+        input->desc().dimsOrder() == DimsOrder::CHW || input->desc().dimsOrder() == DimsOrder::HWC) {
+        const auto ic = input->desc().dim(Dim::C);
+        const auto in = (input->desc().numDims() == 3) ? 1 : input->desc().dim(Dim::N);
+
+        const auto oc = output->desc().dim(Dim::C);
+        const auto on = (output->desc().numDims() == 3) ? 1 : output->desc().dim(Dim::N);
+
+        auto padsBegin = _layer->GetParamAsInts("pads_begin", {});
+        auto padsEnd = _layer->GetParamAsInts("pads_end", {});
+
+        const auto isPadZeros = [](const std::vector<int>& pad) {
+            return std::all_of(pad.begin(), pad.end(), [](int i) { return i == 0; });
+        };
+
+        if (ic == oc && in == 1 && on == 1 && isPadZeros(padsBegin) && isPadZeros(padsEnd)) {
+            ie::details::CaselessEq<std::string> cmp;
+            if (cmp(interpolateMode, "nearest")) {
+                // current "Resample" supports the following "Interpolate" modes only:
+                // coordinate_transformation_mode = half_pixel; nearest_mode = round_prefer_ceil;
+                // other "Interpolate" modes are translated to the default ones
+                const auto antialias = _layer->GetParamAsBool("antialias", false);
+                _stageBuilder->addResampleNearestStage(model,
+                                                       _layer->name,
+                                                       _layer,
+                                                       antialias,
+                                                       -1.0f,
+                                                       input,
+                                                       output);
+            } else if (cmp(interpolateMode, "linear")) {
+                // current "Interp" supports modes "align_corners" and "asymetric" only
+                const auto coordinate_transformation_mode = _layer->GetParamAsString("coordinate_transformation_mode", "half_pixel");
```

Comment:
What if `half_pixel` comes? Is there any assert?

Reply:
It will be processed as asymmetric mode. For some cases it is ok. We should not assert these cases, they should be processed anyway (Probably for some cases a bit incorrect in some pixels)